misc/windows-deploy/containersetup.ps1

```diff
@@ -0,0 +1,27 @@
+# Copyright 2014-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the
+# License is located at
+#
+#	http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+$CREDENTIAL_ADDRESS = "169.254.170.2"
+$EC2_METADATA_ADDRESS = "169.254.169.254"
+
+# Clear pre-existing routes that may conflict
+Get-NetRoute -DestinationPrefix "$($CREDENTIAL_ADDRESS)/32" -PolicyStore PersistentStore -Erroraction:Ignore | Remove-NetRoute -Confirm:$false -Erroraction:Ignore
+Get-NetRoute -DestinationPrefix "$($CREDENTIAL_ADDRESS)/32" -PolicyStore ActiveStore -Erroraction:Ignore | Remove-NetRoute -Confirm:$false -Erroraction:Ignore
+Get-NetRoute -DestinationPrefix "$($EC2_METADATA_ADDRESS)/32" -PolicyStore PersistentStore -Erroraction:Ignore | Remove-NetRoute -Confirm:$false -Erroraction:Ignore
+Get-NetRoute -DestinationPrefix "$($EC2_METADATA_ADDRESS)/32" -PolicyStore ActiveStore -Erroraction:Ignore | Remove-NetRoute -Confirm:$false -Erroraction:Ignore
+
+# Update routes
+[string]$gateway = (Get-WMIObject -Class Win32_IP4RouteTable | Where { $_.Destination -eq '0.0.0.0' -and $_.Mask -eq '0.0.0.0' } | Sort-Object Metric1 | Select NextHop).NextHop
+[int]$ifIndex = (Get-NetAdapter -InterfaceDescription "Hyper-V Virtual Ethernet*" | Sort-Object | Select -First 1).ifIndex
+New-NetRoute -DestinationPrefix "$($CREDENTIAL_ADDRESS)/32" -InterfaceIndex $ifindex -NextHop $gateway

```

Comment:
I dont think this is same as the original command which this has replaced. The original line was route -p add 169.254.170.2 mask 255.255.255.255 0.0.0.0. I think this should be replaced with the following if you want to use this cmdlet.

New-NetRoute -DestinationPrefix 169.254.170.2/32 -InterfaceIndex $ifIndex -NextHop 0.0.0.0

I confirmed that the above works, but specifying $gateway for this doesnt work. Please note that the two features work differently: One talks to a listening port on the host, while the other is routed to a listening port in dom0 (the metadata server).

Reply:
This worked for me. Perhaps my testing was invalid because of something specific that I did. Would you mind sharing your repro steps so I can test this failure case that you hit?

Reply:
@tehshwann Did you ever get this tested with @khannag's test cases?

Reply:
@tehshwann Any update here?